pkgs/build-support/make-darwin-bundle/write-darwin-bundle.nix

```diff
@@ -17,23 +17,30 @@ in writeScriptBin "write-darwin-bundle" ''
 
     readonly prefix=$1
     readonly name=$2
-    readonly exec=$3
+    readonly execName=''${3%% *} # Before the first space
+    [[ $3 =~ " " ]] && readonly execArgs=''${3#* } # Everything after the first space
```

Comment:
`Exec` entries can have spaces if they’re quoted. I wonder if we’re handling quoting properly at all currently.

Reply:
I intentionally chose to not deal with `Exec` entries that had spaces for the sake of simplicity, as none of the the `.desktop` files I had checked used spaces in the executable name

Reply:
Could that be documented with a comment in the script itself?

Reply:
Updated :+1: